Mathlib/AlgebraicGeometry/Curve/Smooth.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2024 Fangming Li. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fangming Li
+-/
+import Mathlib.AlgebraicGeometry.Scheme
+import Mathlib.RingTheory.Smooth.Basic
+
+/-!
+# Smooth morphisms of schemes
+
+In this file, we define when a morphism of schemes is smooth.
+-/
+
+/--
+Smooth homomorphisms between rings.
+-/
+def RingHom.IsSmooth {R A : Type _} [CommRing R] [CommRing A] (f : R →+* A) : Prop :=
+  @Algebra.Smooth R _ A _ (RingHom.toAlgebra f)
+
+namespace AlgebraicGeometry
+
+section ClosedImmersion
+
+/--
+Closed immersions between locally ringed spaces.
+-/
+class LocallyRingedSpace.IsClosedImmersion {X : LocallyRingedSpace} {Y : LocallyRingedSpace}
+    (f : X ⟶ Y) : Prop where
+  val_base_closed : ClosedEmbedding f.val.base
+  epi : ∀ (x : Y.toPresheafedSpace),
+    Function.Surjective ((TopCat.Presheaf.stalkFunctor CommRingCat x).map f.val.c)
+
+/--
+Closed immersions between locally ringed spaces.
+-/
+abbrev IsClosedImmersion {X Y : Scheme} (f : X ⟶ Y) : Prop :=
+  LocallyRingedSpace.IsClosedImmersion f
+
```

Comment:
We essentially have this as https://leanprover-community.github.io/mathlib4_docs/Mathlib/AlgebraicGeometry/Morphisms/ClosedImmersion.html#AlgebraicGeometry.IsClosedImmersion and I don't think we necessarily need it for `LocallyRingedSpace`s, if you want to introduce it, then please open a separate PR for it.